Sources/Transaction/SignedContingentInputBuilderUtils.swift

```diff
@@ -0,0 +1,213 @@
+                    switch $0.errorCode {
+                    case .invalidInput:
+                        return .invalidInput("\(redacting: $0.description)")
+                    case .attestationVerificationFailed:
+                        return .attestationVerificationFailed("\(redacting: $0.description)")
+                    default:
+                        // Safety: mc_signed_contingent_input_builder_add_required_output should not
+                        // throw non-documented errors.
+                        logger.fatalError("Unhandled LibMobileCoin error: \(redacting: $0)")
+                    }
+                }
+            }
+        }
+
+        return publicAddress.withUnsafeCStructPointer { publicAddressPtr in
+            withMcRngObjCallback(rng: rng) { rngCallbackPtr in
+                // Using an in-line func for the FFI Inner to keep closure size low.
+                ffiInner(publicAddressPtr: publicAddressPtr, rngCallbackPtr: rngCallbackPtr)
+            }
+        }.map { txOutData in
+            guard let txOut = TxOut(serializedData: txOutData) else {
+                // Safety: mc_signed_contingent_input_builder_add_required_output should always
+                // return valid data on success.
+                logger.fatalError("mc_signed_contingent_input_builder_add_required_output " +
+                    "returned invalid data: \(redacting: txOutData.base64EncodedString())")
+            }
+
+            return txOut
+        }
+    }
+
+    static func addRequiredChangeOutput(
+        ptr: OpaquePointer,
+        accountKey: AccountKey,
+        amount: Amount,
+        rng: MobileCoinRng
+    ) -> Result<TxOut, TransactionBuilderError> {
+        var confirmationNumberData = Data32()
+
+        let result: Result<Data, TransactionBuilderError> = McAccountKey.withUnsafePointer(
+            viewPrivateKey: accountKey.viewPrivateKey,
+            spendPrivateKey: accountKey.spendPrivateKey,
+            fogInfo: accountKey.fogInfo
+        ) { accountKeyPtr in
+            withMcRngObjCallback(rng: rng) { rngCallbackPtr in
+                confirmationNumberData.asMcMutableBuffer { confirmationNumberPtr in
+
+                    Data.make(withMcDataBytes: { errorPtr in
+                        mc_signed_contingent_input_builder_add_required_change_output(
+                            accountKeyPtr,
+                            ptr,
+                            amount.value,
+                            amount.tokenId.value,
+                            rngCallbackPtr,
+                            confirmationNumberPtr,
+                            &errorPtr)
+                    }).mapError {
+                        switch $0.errorCode {
+                        case .invalidInput:
+                            return .invalidInput("\(redacting: $0.description)")
+                        case .attestationVerificationFailed:
+                            return .attestationVerificationFailed(
```

Comment:
same here, not sure if this can be thrown

Reply:
removed attestationVerificationFailed case